frontend/src/main/scala/bloop/engine/tasks/Tasks.scala

```diff
@@ -112,53 +128,43 @@ object Tasks {
       testEventHandler: BloopTestSuiteEventHandler,
       runInParallel: Boolean = false,
       mode: RunMode
-  ): Task[State] = {
-    import state.logger
-    implicit val logContext: DebugFilter = DebugFilter.Test
+  ): Task[TestRuns] = {
 
-    var failure = false
     val testTasks = projectsToTest.map { project =>
-      /* Intercept test failures to set the correct error code */
-      val failureHandler = new LoggingEventHandler(state.logger) {
+      val resultsBuilder = List.newBuilder[TestSuiteEvent.Results]
+      val handleAndStore = new LoggingEventHandler(state.logger) {
```

Comment:
I'd add explicit warning/information that `resultsBuilder` will be changed by event handler (`handleAndStore`) during test run but it's safe to use that because `result` is called after test run is finished - in TestTask.runTestSuites(...).map(...)